File: webui/src/types/template.ts
export class Template {
  id: number
  name: string
  content: string
  description: string | null
  settings: object | null
  isPublic: boolean

  constructor(
    id: number,
    name: string,
    content: string,
    description: string | null = null,
    settings: object | null = null,
    isPublic: boolean = false,
  ) {
    this.id = id
    this.name = name
    this.content = content
    this.description = description
    this.settings = settings
    this.isPublic = isPublic
  }
}

File: src/Models/Template.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace LLMRolePlay.Models
{
  public partial class DBContext : Microsoft.EntityFrameworkCore.DbContext
  {
    public DbSet<Template> Templates { get; set; }
  }

  public class Template
  {
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public uint Id { get; set; }
    public required string Name { get; set; }
    public required string Content { get; set; }
    public string? Description { get; set; }
    public bool IsPublic { get; set; }
    public uint UserId { get; set; }
    public string? Settings { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
  }

}